src/paginationCalls/pageDescriptor.ts

```diff
@@ -103,6 +114,110 @@ export class PageDescriptor implements Descriptor {
     return stream;
   }
 
+  /**
+   * Create an async iterable which can be recursively called for data on-demand.
+   */
+  asyncIterate(
+    apiCall: GaxCall,
+    request: RequestType,
+    options: CallSettings
+  ): AsyncIterable<{} | undefined> {
+    const iterable = this.createIterator(options);
+    const funcPromise =
+      typeof apiCall === 'function' ? Promise.resolve(apiCall) : apiCall;
+    funcPromise
+      .then((func: GaxCall) => {
+        this.resolveParams(request, func, options);
+      })
+      .catch(error => {
+        throw new Error(error);
+      });
+    return iterable;
+  }
+
+  createIterator(options: CallSettings): AsyncIterable<{} | undefined> {
+    const responsePageTokenFieldName = this.responsePageTokenField;
+    const requestPageTokenFieldName = this.requestPageTokenField;
+    let resolveRequest = this.resolveRequest;
+    let resolveFunction = this.resolveFunction;
+    const asyncIterable = {
+      [Symbol.asyncIterator]() {
+        const funcPromise = new Promise((resolve, reject) => {
+          resolveFunction = resolve;
+        });
+        const requestPromise = new Promise((resolve, reject) => {
+          resolveRequest = resolve;
+        });
+        const cache: Array<{}> = [];
+        let nextPageRequest: RequestType = {};
+        let firstCall = true;
+        return {
+          async next() {
+            const ongoingCall = new call.OngoingCallPromise(options.promise);
+
+            const func = (await funcPromise) as SimpleCallbackFunction;
+            const request = (await requestPromise) as RequestType;
+            if (firstCall) {
+              ongoingCall.call(func, request);
+              const [
+                response,
+                nextRequest,
+                rawresponse,
+              ] = await ongoingCall.promise;
+              //@ts-ignore
```

Comment:
could we create an interface for `responses` sot hat we can avoid the `@ts-ignore`.

Reply:
thanks for the feedback, yes, will give the responses type and remove the annotation!